lib/synapse/service_watcher/base.rb

```diff
@@ -95,13 +100,44 @@ def validate_discovery_opts
     end
 
     def set_backends(new_backends)
-      if @keep_default_servers
-        @backends = @default_servers + new_backends
+      # Aggregate and deduplicate all potential backend service instances.
```

Comment:
are you really concerned about duplicates in the default servers list? if that's the case, maybe we should de-dup `@default_servers`  when we set it